siliconcompiler/schema/schema_cfg.py

```diff
@@ -3506,6 +3484,177 @@ def schema_constraint(cfg):
+            switch="-server_memory <str>",
+            example= ["cli: -server_memory 8000",
+                    "api: chip.set('server', 'memory', '8000')"],
+            schelp="""
+            Specifies the amount of memory required to run the job,
+            specified in MB. For the slurm scheduler, this translates to
+            the '--mem' switch. """)
+
+    scparam(cfg, ['server', 'constraint'],
+            sctype='str',
+            scope='job',
+            shorthelp="Server: Machine constraint",
+            switch="-server_constraint <str>",
+            example= [
+                "cli: -server_constraint 'riscv&gpu'",
+                "api: chip.set('server', 'constraint', 'riscv&gpu')"],
+            schelp="""
+            Provides machine feature constraints to the job scheduler for
+            selecting compute nodes. Feature names are strings assigned to
+            computes nodes by the job scheduler administrator. The constraint
+            parameter is passed through to the job schedular unmodified. For
+            the slurm scheduler, this translates to the '--constraint'
+            switch. See the slurm documentation for examples of the types of
+            constraints supported.""")
+
+    scparam(cfg, ['server', 'queue'],
+            sctype='str',
+            scope='job',
+            shorthelp="Server: Queue selection",
+            switch="-server_queue <str>",
+            example= ["cli: -server_queue nightrun",
+                      "api: chip.set('server', 'queue', 'nightrun')"],
+            schelp="""
+            Schedule the job for the specified queue. With slurm, this
+            translates to 'partition'. The queue name must match
+            the name of an existing job schemduler queue.""")
+
+    scparam(cfg, ['server', 'nice'],
+            sctype='int',
+            scope='job',
+            shorthelp="Server: Scheduling priority",
+            switch="-server_nice <int>",
+            example= ["cli: -server_nice 100",
+                    "api: chip.set('server', 'nice', '100')"],
+            schelp="""
+            Specifies the scheduling priority for the job. Negative numbers
+            increases priority while positive numbers decrease priority.
+            For more information, see the job scheduler documentation.""")
+
+    scparam(cfg, ['server', 'timeout'],
+            sctype='str',
+            scope='job',
+            shorthelp="Server: Timeout value",
+            switch="-server_timeout <str>",
+            example= ["cli: -server_timeout 16:00",
+                    "api: chip.set('server', 'timeout', '16:00')"],
+            schelp="""
+            Specifies the limit on the total job run time.
+            Acceptable slurm time formats include "minutes", "minutes:seconds",
+            "hours:minutes:seconds", "days-hours", "days-hours:minutes",
+            and "days-hours:minutes:seconds".""")
+
+    scparam(cfg, ['server', 'interactive'],
```

Comment:
May be tricky to use this argument with SC since most of our current flows are designed around batch jobs. Could be a nice to have which can be added as we bring interactive workloads into the framework (analog, etc)

Reply:
Having the ability to get access to a large work station on the fly for interactive digital work/gds viewing or analog view is a fundamental need. Getting this feature to work will be a bit tricky, but it's definitely a must have.